### PR TITLE
DevEx/CI improvements

### PR DESCRIPTION
**Pull request by ytham**
- adds additional console logging so users know where the CLI is looking for input/compiled/proven/etc files
- updates the CI to build and use local package links so no more optimistic publishing is required
- updates interface names in `react` to be in line with `client` names
- adds additional exports in `client` so that users do not need to import lower-layer axiom npm packages

### CODE REVIEW

**Comment by ytham** [comment]
PR description updated and ready for review @yi-sun @jonathanpwang

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -48,13 +48,17 @@ jobs:
...
-        working-directory: ./client
-        run: npm install && npm run build
+      - name: Set all packages to local and build
+        working-directory: ./
+        run: npm run local ci
 
-      - name: Install Harness dependencies
-        working-directory: ./harness
-        run: npm install
+      # - name: Build Client
```
Comment (yi-sun) [inline .github/workflows/test.yml:55]: Should we remove these?

File: react/src/AxiomCircuitProvider.tsx [hunk 2]
```diff
@@ -43,20 +43,14 @@ const useAxiomCircuit = <T,>(): AxiomCircuitContextType<T> => {
...
-  build: {
-    vk: string,
-    config: CircuitConfig,
-    querySchema: string,
-    inputSchema: string,
-    circuit: string,
-  }
-  providerUri: string,
+  compiledCircuit: AxiomV2CompiledCircuit,
+  provider: string,
```
Comment (yi-sun) [inline react/src/AxiomCircuitProvider.tsx:53]: Should `provider` be first in this second listing?

File: readme.md [hunk 3]
```diff
@@ -5,9 +5,11 @@ This repository is split into 3 components:
 - `client`: Contains the Axiom CLI interface and developer-facing AxiomCircuit exports
 - `circuit`: Handles lower-level interface with Wasm circuit libraries
 - `harness`: Test harness for `client`
+- `react`: Axiom React component for proving in React/Next.js dApps
```
Comment (yi-sun) [inline readme.md:8]: Should we call this "React component" or "React hook"?

**Review by yi-sun — APPROVED** [review]
Just some small comments, OK to merge after addressing

File: readme.md (hunk 3 above)
Comment (ytham) [reply readme.md:8]: it technically exports a react component as well as a hook (for interacting w/ the component); will update the description accordingly

**Pull request merged**